common/djangoapps/enrollment/views.py

```diff
@@ -26,7 +29,11 @@
     OAuth2AuthenticationAllowInactiveUser,
     SessionAuthenticationAllowInactiveUser
 )
-from openedx.core.lib.api.permissions import ApiKeyHeaderPermission, ApiKeyHeaderPermissionIsAuthenticated
+from openedx.core.lib.api.permissions import (
+    ApiKeyHeaderPermission,
+    ApiKeyHeaderPermissionIsAuthenticated,
+    IsUserInUrlOrStaff
```

Comment:
Actually that class also considers usernames in the URL...hrm...still seems like that opens the door to trouble... 😄